analyzer_test.go

```diff
@@ -198,263 +122,111 @@ func testAnalyzer(t *testing.T, when spec.G, it spec.S) {
-						got := h.MustReadFile(t, path)
-						want := "[metadata]\n  some-layer-key = \"some-layer-value\""
+					it("analyzes layers", func() {
+						mockLayerAnalyzer.EXPECT().RetrieveMetadataFrom(testCache)
+						mockLayerAnalyzer.EXPECT().Analyze(analyzer.Buildpacks, analyzer.SkipLayers, gomock.Any(), gomock.Any())
 
-						h.AssertStringContains(t, string(got), want)
-					})
-					it("does not restore layers for undetected buildpacks", func() {
 						_, err := analyzer.Analyze(image, testCache)
 						h.AssertNil(t, err)
-
-						h.AssertPathDoesNotExist(t, filepath.Join(layerDir, "metadata.buildpack"))
-						h.AssertPathDoesNotExist(t, filepath.Join(layerDir, "escaped_buildpack_id"))
 					})
-				})
-			})
-
-			when("skip-layers is true", func() {
-				it.Before(func() {
-					analyzer.SkipLayers = true
-				})
-
-				it("should return the analyzed metadata", func() {
-					md, err := analyzer.Analyze(image, testCache)
-					h.AssertNil(t, err)
-
-					h.AssertEq(t, md.Image.Reference, "s0m3D1g3sT")
-					h.AssertEq(t, md.Metadata, appImageMetadata)
-				})
 
-				it("does not write buildpack layer metadata", func() {
-					_, err := analyzer.Analyze(image, testCache)
-					h.AssertNil(t, err)
+					when("skip-layers is true", func() {
+						it.Before(func() {
+							analyzer.SkipLayers = true
+							mockLayerAnalyzer.EXPECT().RetrieveMetadataFrom(testCache)
+							mockLayerAnalyzer.EXPECT().Analyze(analyzer.Buildpacks, analyzer.SkipLayers, gomock.Any(), gomock.Any())
+						})
 
-					files, err := ioutil.ReadDir(layerDir)
-					h.AssertNil(t, err)
-					h.AssertEq(t, len(files), 2)
+						it("should return the analyzed metadata", func() {
+							md, err := analyzer.Analyze(image, testCache)
+							h.AssertNil(t, err)
 
-					files, err = ioutil.ReadDir(filepath.Join(layerDir, "metadata.buildpack"))
-					h.AssertNil(t, err)
-					//expect 1 file b/c of store.toml
-					h.AssertEq(t, len(files), 1)
-
-					files, err = ioutil.ReadDir(filepath.Join(layerDir, "no.cache.buildpack"))
-					h.AssertNil(t, err)
-					//expect 1 file b/c of store.toml
-					h.AssertEq(t, len(files), 1)
-				})
-
-				it("restores each store metadata", func() {
-					_, err := analyzer.Analyze(image, testCache)
-					h.AssertNil(t, err)
-					for _, data := range []struct{ name, want string }{
-						// store.toml files.
-						{"metadata.buildpack/store.toml", "[metadata]\n  [metadata.metadata-buildpack-store-data]\n    store-key = \"store-val\""},
-						{"no.cache.buildpack/store.toml", "[metadata]\n  [metadata.no-cache-buildpack-store-data]\n    store-key = \"store-val\""},
-					} {
-						got := h.MustReadFile(t, filepath.Join(layerDir, data.name))
-						h.AssertStringContains(t, string(got), data.want)
-					}
+							h.AssertEq(t, md.Image.Reference, "s0m3D1g3sT")
+							h.AssertEq(t, md.Metadata, appImageMetadata)
+						})
+					})
 				})
 			})
 		})
 
-		when("image is not found", func() {
+		when("platform API < 0.6", func() {
```

Comment:
I propose moving all `when`s concerning earlier platform apis to the bottom of the file, it would make it easier to understand what is specific to which api.